### PR TITLE
chore: a standalone keys implementation for the legacy library

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

With all of the previous PRs out of the way, we can now extract `PublicKey` and `Keypair` into a sub-package without circular dependencies.

The only material change here is that we lazily attach `PublicKey` to the global Borsh schema if you import only `@solana/keys` and nothing else. We do so so that we can still mark the thing as `sideEffects: false`.

### CODE REVIEW

**Comment by steveluscher** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1928** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1928?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1929** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1930** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1930?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1931** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1931?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1932** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1932?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #1933** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1933?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #1934** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1934?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #1935** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1935?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #1936** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment).

**Comment by steveluscher** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1928** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1928?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1929** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1930** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1930?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1931** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1931?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1932** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1932?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #1933** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1933?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #1934** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1934?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #1935** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1935?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #1936** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment).

**Comment by steveluscher** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1928** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1928?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1929** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1930** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1930?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1931** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1931?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1932** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1932?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #1933** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1933?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #1934** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1934?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #1935** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1935?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #1936** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1936?utm_source=stack-comment).

File: packages/legacy/keys/package.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
+    "name": "@solana/keys-legacy",
```
Comment (lorisleiva) [inline packages/legacy/keys/package.json:2]: Let's keep a consistent naming convention. 🙏 

```suggestion
    "name": "@solana/legacy-keys",
```

Comment (steveluscher) [reply packages/legacy/keys/package.json:2]: The idea here is to publish this legacy keys-related stuff at `1.x` and the new stuff at `>2.x`.

Comment (steveluscher) [reply packages/legacy/keys/package.json:2]: Oh, the thing that's not obvious here is that the [build script](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1932/chore-create-custom-publish-strategy-for-legacy-packages?review=PRR_kwDOCLAddc5pYnKy) strips the `-legacy` suffix at build time. The package _actually_ gets published as `@solana/keys` on npm.

Comment (buffalojoec) [reply packages/legacy/keys/package.json:2]: Ohhhh god, I didn't realize this was what you were doing with these. So this is `@solana/keys@1.x` and the experimental one is `@solana/keys@2.x`?

I guess it kind of makes sense though, it aligns with `@solana/web3.js@1.x`. Does that mean someone who wants to use `SubtleCrypto` and `@solana/keys@2.x` with the legacy Web3.js is smoked?

Comment (steveluscher) [reply packages/legacy/keys/package.json:2]: > Does that mean someone who wants to use SubtleCrypto and @solana/keys@2.x with the legacy Web3.js is smoked?

**Pull request closed without merging**